US312 Add map and reintroduce confirm selection page ...

Pheonnexx opened the pull request:
... amended small defect for link on the display title page

Pheonnexx committed:
Reintroduce confirm selection page, amended small defect for link on the display title page

Pheonnexx committed:
Reintroduce confirm selection page, amended small defect for link on the display title page

RichardProsser commented on service/templates/confirm_selection.html:
```diff
@@ -2,58 +2,80 @@
+            {% if params['title']['address_lines'] %}
+              {{ '<br>'.join(params['title']['address_lines'])|safe }}
+            {% else %}
+              {{ _('Not Available') }}
```
Why no `<br>` after the 'else' as well?

Pheonnexx replied:
The <br> is added to make each line of the address go onto a new line - when there is no address there is no need to do this?

RichardProsser commented on service/templates/confirm_selection.html:
```diff
@@ -2,58 +2,80 @@
+          <p>The summary can tell you</p>
+          <ul class="list-bullet">
+            <li>owner's name and address</li>
+            <li>what they paid for the property – you won’t get this if it was bought before April 2000, if no money changed hands, or if the owner didn’t buy it outright</li>
```
"the price paid for the property"

I suggest.

Pheonnexx replied:
I am following the prototype.

RichardProsser commented on service/templates/confirm_selection.html:
```diff
@@ -2,58 +2,80 @@
+          </div>
+          <hr>
+          <h2 class="heading-medium">You won't get</h2>
+          <p>You'll need to <a target="_blank" href="https://www.gov.uk/get-information-about-property-and-land">use a different Land Registry service</a> if you need more detailed information like</p>
```
"Link opens in new window" perhaps?

RichardProsser commented on service/templates/display_title.html:
```diff
@@ -174,7 +174,7 @@ <h5 class="expand-bottom-half {% if loop.index != 1 %} expand-top-half {% endif
     <h6 class="heading-medium">{{ _('What next') }}</h6>
     <ul>
-      <li><a href="/drv-15/search">{{ _('Search for another title using this service') }}</a></li>
+      <li><a href="{{ url_for('find_titles') }}">{{ _('Search for another title using this service') }}</a></li>
```
If I remember rightly the 'find_titles' link is a bit ambiguous - the behaviour varies according to whether the action is Get or Post. This usage is probably OK however.

Pheonnexx committed:
Updated unit tests

Modified English/Welsh links, now display on all pages (layout.html), refactored css to display appropriately.

Refactored css following comment on merge request

Refactored css following comment on merge request

Add explanation if caution title in search results

Add sidebar for help on search results

Change the FaP link to the shorter prettier version

Added id to help sidebar

Pheonnexx committed:
WIP changed to use new proto design

Pheonnexx committed:
WIP changed to use new proto design

Pheonnexx committed:
Amend styling on maps and link javascript correctly

Pheonnexx committed:
Sort out freehold and leasehold text on page

Pheonnexx committed:
rebase

Pheonnexx committed:
remove a merge issue

Pheonnexx committed:
fix merge defect

Pheonnexx committed:
fix merge defect with welsh translation

russwillis commented on config.py:
```diff
@@ -14,7 +14,7 @@
 show_full_title_pdf = os.environ['SHOW_FULL_TITLE_PDF'].lower() == 'true'
-title_register_summary_price = "&pound;1.20 (incl. VAT)"
+title_register_summary_price = "£2.00 (incl. VAT)"
 
```
I think we need to make the pound sign HTML safe again.

russwillis commented on service/templates/confirm_selection.html:
```diff
@@ -2,58 +2,133 @@
+        If you want to keep a copy you can print it out</p>
+    </div>
+
+      <h2 class="heading-medium">The summary costs £2</h2>
```
aha. This is why your tests work?
Should be using the variable { params.price }

russwillis commented on service/templates/display_title.html:
```diff
@@ -1,9 +1,13 @@
+    cursor: pointer;
+    text-decoration: underline;
+    padding-left:0;
   }
```
why have we got style in the html rather than the CSS?

russwillis commented on service/templates/display_title.html:
```diff
@@ -136,7 +140,7 @@ <h5 class="expand-bottom-half {% if loop.index != 1 %} expand-top-half {% endif
 
   <div class="column-third">
-    <div id="map">
+    <div id="map" class="polygon" style="height: 0 !important; padding-bottom: 100% !important;width: 100% !important;">
```
again. Inline style? Should be in the CSS.

Pheonnexx replied:
Kept until we refactor the css/html of the frontend

russwillis commented on service/templates/display_title.html:
```diff
@@ -172,8 +176,8 @@ <h6 class="heading-medium">{{ _('What next') }}</h6>
 <script type='text/javascript'>
-  //var indexData = {"crs": {"properties": {"name": "urn:ogc:def:crs:EPSG::27700"}, "type": "name"}, "geometry": {"coordinates": [[[441187.5, 112344.0], [441186.0, 112354.0], [441193.0, 112356.5], [441194.5, 112346.5]]], "type": "Polygon"}, "properties": {"colour": 25, "feature_id": 4019, "graphic_type": "Bordered Polygon", "render_attributes": {"border_colour": 25, "border_width": 0, "exterior_e ...
-  var indexData = {{ title.indexPolygon | tojson }};
+  var indexData = {"crs": {"properties": {"name": "urn:ogc:def:crs:EPSG::27700"}, "type": "name"}, "geometry": {"coordinates": [[[441187.5, 112344.0], [441186.0, 112354.0], [441193.0, 112356.5], [441194.5, 112346.5]]], "type": "Polygon"}, "properties": {"colour": 25, "feature_id": 4019, "graphic_type": "Bordered Polygon", "render_attributes": {"border_colour": 25, "border_width": 0, "exterior_edg ...
```
This is hardcoded surely? Why comment out the next line which appears to point to the dynamic version?

Pheonnexx committed:
Amend code due to code review comments

Pheonnexx committed:
Introduce css for the map in the summary page
